libs/pages/application/src/lib/feature/page-settings-deployment-restrictions-feature/page-settings-deployment-restrictions-feature.tsx

```diff
@@ -19,10 +24,43 @@ import {
 import CrudModalFeature from './crud-modal-feature/crud-modal-feature'
 
 export function PageSettingsDeploymentRestrictionsFeature() {
-  const { projectId = '', environmentId = '', applicationId = '' } = useParams()
+  const { projectId, environmentId, applicationId } = useParams()
+  const { data: serviceType } = useServiceType({ environmentId, serviceId: applicationId })
+  if (
+    !projectId ||
+    !environmentId ||
+    !applicationId ||
+    !serviceType ||
+    !(isApplicationType(serviceType) || isJobType(serviceType))
+  ) {
+    return null
```

Comment:
Not sure to understand why you separate it, and it provides a white section when it's the first time you're on it, do you have some details?

And this check is already done on `<PageSettingsFeature />` not sure if it's necessary to have it